### PR TITLE
fix: Clear database and token cache on different user login

### LINKED ISSUE #210: Clear all the database on new user login
Currently we are keeping database excluding `User` and `UserDetail`tables on log out. Which is problem for new login as it still shows old data for new user.

Correct approach should be keeping all the database on logout. On login if data exist check for the user id. If user id of the new user doesn't matches with existing id then clear all the database otherwise keep the same.

@iamareebjamal you may want to take up this.

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #210 

- Clear database on new user login
- Invalidate token cache from appropriate locations
- Add test

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@mejariamol @arp95 @Shailesh351 @rohanagarwal94 Please review

**Comment by mariobehling** [comment]
Please resolve conflicts.

File: app/src/main/java/org/fossasia/openevent/app/data/UtilModel.java [hunk 1]
```diff
@@ -44,18 +49,33 @@ public void saveString(String key, String value) {
...
         saveString(Constants.SHARED_PREFS_TOKEN, token);
     }
 
     @Override
     public boolean isConnected() {
         return NetworkUtils.isNetworkConnected(context);
     }
+
+    @Override
+    public Completable deleteDatabase() {
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/data/UtilModel.java]: Can we test this if it is actually deleting database?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/UtilModel.java]: Unit test or manual testing. For manual testing, you can check stetho

**Pull request merged**